test/unity/toolkits/drawing_classifier/test_dc_data_iterator.cxx

```diff
@@ -39,114 +40,200 @@ const std::vector<std::string> unique_labels = {"foo", "bar", "baz"};
+  for (size_t i = 0; i < actual_dim; i++) {
     TS_ASSERT_EQUALS(actual_shape[i], expected_shape[i]);
   }
+
   gl_sframe data = params.data;
   size_t index_in_data = 0;
   /* Test target contents */
   const float *actual_target_data = next_batch.targets.data();
-  for (size_t index_in_batch = 0; index_in_batch < batch_size; index_in_batch++) {
+  for (size_t index_in_batch = 0; index_in_batch < batch_size;
+       index_in_batch++) {
     float expected_target = static_cast<float>(
-      class_to_index_map[
-        data[params.target_column_name][index_in_data].to<flex_string>()
-      ]);
+        class_to_index_map[data[params.target_column_name][index_in_data]
+                               .to<flex_string>()]);
     float actual_target = actual_target_data[index_in_batch];
     TS_ASSERT_EQUALS(expected_target, actual_target);
     index_in_data = (index_in_data + 1) % data.size();
   }
+
   /* Test drawing contents */
   index_in_data = 0;
   const float *actual_drawing_data = next_batch.drawings.data();
-  for (size_t index_in_batch = 0; index_in_batch < batch_size; index_in_batch++) {
+  for (size_t index_in_batch = 0; index_in_batch < batch_size;
+       index_in_batch++) {
     flex_image decoded_drawing = image_util::decode_image(
-    data[params.feature_column_name][index_in_data].to<flex_image>());
-    const unsigned char *expected_drawing_data = decoded_drawing.get_image_data();
+        data[params.feature_column_name][index_in_data].to<flex_image>());
+    const unsigned char *expected_drawing_data =
+        decoded_drawing.get_image_data();
     for (size_t row = 0; row < IMAGE_HEIGHT; row++) {
       for (size_t col = 0; col < IMAGE_WIDTH; col++) {
         // Asserting that the (row, col) index of every drawing in the batch
         // matches the (row, col) index we have in the original SFrame.
         TS_ASSERT_EQUALS(
-          static_cast<unsigned char>(actual_drawing_data[
-            index_in_batch * IMAGE_WIDTH * IMAGE_HEIGHT * 1
-            + row * IMAGE_WIDTH * 1
-            + col * 1]),
-          expected_drawing_data[row * IMAGE_WIDTH + col]
-          );
+            static_cast<unsigned char>(
+                actual_drawing_data[index_in_batch * IMAGE_WIDTH *
+                                        IMAGE_HEIGHT * 1 +
+                                    row * IMAGE_WIDTH * 1 + col * 1]),
+            expected_drawing_data[row * IMAGE_WIDTH + col]);
       }
     }
     index_in_data = (index_in_data + 1) % data.size();
   }
 }
 
 BOOST_AUTO_TEST_CASE(test_simple_data_iterator) {
-  static constexpr size_t MAX_NUM_ROWS = 4;
-  static constexpr size_t MAX_BATCH_SIZE = 8;
+  constexpr size_t MAX_NUM_ROWS = 4;
+  constexpr size_t MAX_BATCH_SIZE = 8;
+
   for (size_t num_rows = 1; num_rows <= MAX_NUM_ROWS; num_rows++) {
     for (size_t batch_size = 1; batch_size <= MAX_BATCH_SIZE; batch_size++) {
-      drawing_data_generator data_generator(num_rows, unique_labels);
```

Comment:
because `unique_labels` is global variable. We shouldn't use `static` inside of function unless we know the function is **NOT** stateless. `constexpr` is already sufficient to use.